src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,97 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] hashMap;
+
+    public MyHashMap() {
+        size = 0;
+        hashMap = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if ((LOAD_FACTOR * hashMap.length) <= size) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getIndex(key);
+        if (hashMap[index] == null) {
+            hashMap[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> tempNode = hashMap[index];
+        while (tempNode.next != null) {
+            if (tempNode.key == key || tempNode.key != null && key.equals(tempNode.key)) {
+                tempNode.value = value;
+                return;
+            }
+            tempNode = tempNode.next;
+        }
+        if (tempNode.key == key || tempNode.key != null && tempNode.key.equals(key)) {
+            tempNode.value = value;
+            return;
+        }
+        tempNode.next = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return hashMap[0].value;
+        }
+        int index = getIndex(key);
+        Node<K, V> tempNode = hashMap[index];
+        while (tempNode != null) {
+            if (tempNode.key.equals(key)) {
+                return tempNode.value;
+            }
+            tempNode = tempNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode()) % (hashMap.length) + 1;
```

Comment:
Are you trying to leave zero bucket only for null-key? But why? I am very surprised that this code has passed all the tests without ArrayIndexOutOfBoundException. With +1 there is possible situation when index == hashMap.length

Reply:
you can't get outOfBound, cause of load factor( resize will take place sooner than we will have index == hashMap.length) , but agree that better to remove +1 
use ternary operator instead of if, code will get shorter

Reply:
My idea was to made (hashMap.length - ) + 1; it make possible to put only one null at 0 position, and 15 other for other values, anyway I fixed it.